### PR TITLE
Swift Package Manager support (#1)

### PR DESCRIPTION
**Pull request by aidansteele**
Thanks @nst for agreeing to this so quickly! I am also relatively new to the Swift Package Manager ecosystem, but the [documentation](https://swift.org/package-manager) is reasonably straight forward. 

SPM expects source to be laid out in a certain way and then it will auto-detect all source and test files. Additional files were created by invoking `swift package init`. The xcodeproj was deleted, but a new one can be generated on-demand by executing `swift package generate-xcodeproj`.

The last task is tagging releases. When someone wishes to use `BinUtils` in their project, their `Package.swift` will look something like this:

``` swift
import PackageDescription

let package = Package(
    name: "MyPackage",
    dependencies: [
        .Package(url: "https://github.com/nst/BinUtils.git", majorVersion: 0),
    ]
)
```

In this example I've not specified an exact version, just anything in the 0.x.x series. The package manager clones your repo and checks out the latest tagged release. If there aren't any tags, I'm not sure if it's possible to use the project. You can feel free to choose any version number you want, but Apple's docs (linked above) recommend [semantic versioning](http://semver.org/) - so maybe something like 0.1.0 for the current state of the project?

Let me know if you have any other questions and I'll try to help!

### CODE REVIEW

File: BinUtils/BinUtils/main.swift [hunk 1]
```diff
@@ -1,34 +0,0 @@
-//
```
Comment (aidansteele) [inline BinUtils/BinUtils/main.swift:1]: I wasn't completely sure what to do with this file. Should they be converted to tests?

**Comment by aidansteele** [comment]
I also added `public` access level modifiers to a handful of places. This was because the default access level is `internal`, which isn't very useful for our needs. :)

File: BinUtils/BinUtils/main.swift (hunk 1 above)
Comment (nst) [reply BinUtils/BinUtils/main.swift:1]: No, main.swift just contains quick tests like in a playground.

The real tests are in the BinUtilsTests target.

**Comment by evgeny-sureev** [comment]
You need also add a version tag or else Package Manager unable to fetch dependency.

**Comment by nst** [comment]
I added tag "0.1", is it sufficient?

**Comment by evgeny-sureev** [comment]
Sorry, forget to mention that tags must contain three versions: major.minor.patch, for example "0.1.0".

**Comment by nst** [comment]
ok, should be good now

**Comment by evgeny-sureev** [comment]
Works now, thank you!

**Pull request merged**